example/v5_no_tls_both.cpp

```diff
@@ -311,9 +312,12 @@ void server_proc(Server& s, std::set<con_sp_t>& connections, mi_sub_con& subs) {
                         r.first->con->publish(
                             boost::asio::buffer(topic_name),
                             boost::asio::buffer(contents),
-                            std::min(r.first->qos_value, pubopts.get_qos()) | pubopts.get_retain(),
-                            MQTT_NS::v5::properties{},
-                            std::make_pair(topic_name, contents)
+                            std::min(r.first->qos_value, pubopts.get_qos()) |
+                            (
+                                r.first->rap_value == MQTT_NS::rap::retain ? pubopts.get_retain()
```

Comment:
You're assigning a new value to r.first->rap_value, and then using it?

I think it would be easier for readers of this example code to understand if this was two seperate operations. First an assignment, above the call to publish, and then a read, here in the publish function call.

Reply:
> You're assigning a new value to r.first->rap_value, and then using it?

No. Delivering the received publish retain or not is depends on rap_value. This code refer to rap_value and choose retain value.

But I think it is difficult to understand. Especially this is an example code.
So I remove conditional operator and introduce temporary variable.

Reply:
I understand.

I thought == was just =. My mistake.